onnxruntime/contrib_ops/cuda/bert/attention_softmax.h

```diff
@@ -180,7 +181,16 @@ __device__ inline void SoftmaxWithRawMaskSmall(const int all_sequence_length,
   if (threadIdx.x < all_sequence_length) {
     const int batch_index = blockIdx.y;
     const int sequence_index = blockIdx.x % sequence_length;
-    const int mask_offset = (mask_dimension == 2) ? batch_index * all_sequence_length + threadIdx.x : batch_index * sequence_length * all_sequence_length + sequence_index * all_sequence_length + threadIdx.x;
+    int mask_offset = 0;
+    if (mask_dimension == 2) {
+      mask_offset = batch_index * all_sequence_length + threadIdx.x;
+    } else if (mask_dimension == 3) {
+      mask_offset = batch_index * sequence_length * all_sequence_length + sequence_index * all_sequence_length + threadIdx.x;
```

Comment:
Use the following might be faster (4 operations instead of 5):
```
(batch_index * sequence_length + sequence_index) * all_sequence_length + threadIdex.x
```

For 4D, it could have similar change.